pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -276,6 +288,17 @@ func enrichClusterSystemData(doc *api.OpenShiftClusterDocument, systemData *api.
 	}
 }
 
+func (f *frontend) persistIdentityUrl(cluster *api.OpenShiftCluster, identityURL []string) error {
+	if identityURL == nil {
+		return nil
+	} else if len(identityURL) != 1 {
+		return fmt.Errorf("identityURL cannot contain multiple values: %v", identityURL)
+	}
+	cluster.Identity.IdentityURL = identityURL[0]
```

Comment:
This function relies on identityURL for checking if the cluster is MIWI or not, IMO, MIWI check can be done by a nil-check on `cluster.Identity`.
And error is returned if identityURL is nil or have more than 1 values.
Alternative approach can be one from the [Preview API design](https://docs.google.com/document/d/1AIuf6H9y0Jjboywp0NgD8-9lU1jCj5WrE83jh58rEg4/edit#heading=h.ds2n08h4tc02) too
WDYT?